pvlib/test/test_irradiance.py

```diff
@@ -940,3 +940,53 @@ def test_clearness_index_zenith_independent(airmass_kt):
                                                         airmass)
     expected = pd.Series([np.nan, 0.553744437562], index=times)
     assert_series_equal(out, expected)
+
+
+def test_adjust_direct_for_horizon(irrad_data, ephem_data):
+    zero_horizon = []
+    max_horizon = []
+    for i in range(-180, 181):
+        zero_horizon.append((i, 0.0))
+        max_horizon.append((i, 90.0))
+
+    zero_adjusted = irradiance.adjust_direct_for_horizon(irrad_data["dni"],
+                                                         zero_horizon,
+                                                         ephem_data["azimuth"],
+                                                         ephem_data["zenith"])
+    zero_expected = irrad_data["dni"]
+    assert_allclose(zero_adjusted.values, zero_expected.values)
+
+    max_adjusted = irradiance.adjust_direct_for_horizon(irrad_data["dni"],
+                                                        max_horizon,
+                                                        ephem_data["azimuth"],
+                                                        ephem_data["zenith"])
+
+    max_expected = pd.Series([0, 0, 0, 0])
+    assert_allclose(max_adjusted.values, max_expected.values, atol=1e-7)
+
+
+def test_horizon_adjusted_isotropic():
+    zero_horizon = []
+    max_horizon = []
```

Comment:
I suggest adding a couple of intermediate test cases.

Reply:
done